alot/commands/__init__.py

```diff
@@ -17,11 +17,24 @@ class Command(object):
     repeatable = False
 
     def __init__(self):
-        self.prehook = None
-        self.posthook = None
         self.undoable = False
         self.help = self.__doc__
 
+        class_ = type(self)
+        class_name = class_.__name__
+        cmdname, mode = reverse_lookup_command(class_)
```

Comment:
it seems that you only need the cmd name and mode string to look up the hook here. Why not turn `reverse_lookup_command` into a function `get_hook` that takes `class_` and returns the callable hook or None?